node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -136,35 +137,152 @@ fn collect_externals_from_multi_config(
     )
 }
 
+fn extract_values_vcl_fill_multiconfig_vec(
+    config_file_vcl: Box<dyn VirtualCommandLine>,
+    environment_vcl: Box<dyn VirtualCommandLine>,
+    commandline_vcl: Box<dyn VirtualCommandLine>,
+    user_specific_data: UserSpecifiedData,
+) -> (Vec<String>, Vec<String>) {
+    let mut unspecified_vec: Vec<String> = vec!["".to_string()];
+    let mut specified_vec: Vec<String> = vec!["".to_string()];
+    let config_file_specified = user_specific_data.config_file_spec;
+    let config_file_path = user_specific_data.config_file;
+    let extract_value_from_vcl =
+        |vcl: &dyn VirtualCommandLine, name: &str, var: &str, spec: bool| {
+            let args = vcl.args();
+            let extracted_args = args
+                .iter()
+                .enumerate()
+                .filter(|(_index, vcl_arg)| vcl_arg.deref() == &name.to_string())
+                .collect_vec();
+            match extracted_args.is_empty() {
+                false => {
+                    let val_str = args[extracted_args[0].0 + 1].clone();
+                    (val_str, true)
+                }
+                true => (var.to_string(), spec),
+            }
+        };
+    let (cf_real_user, cf_real_user_specified) = extract_value_from_vcl(
+        &*config_file_vcl,
+        "--real-user",
+        user_specific_data.real_user.to_string().as_str(),
+        user_specific_data.real_user_spec,
+    );
+
+    let (env_real_user, env_real_user_specified) = extract_value_from_vcl(
+        &*environment_vcl,
+        "--real-user",
+        user_specific_data.real_user.to_string().as_str(),
+        user_specific_data.real_user_spec,
+    );
+
+    let (cmd_real_user, cmd_real_user_specified) = extract_value_from_vcl(
+        &*commandline_vcl,
+        "--real-user",
+        user_specific_data.real_user.to_string().as_str(),
+        user_specific_data.real_user_spec,
+    );
+
+    let mut fill_specified_or_unspecified_box =
```

Comment:
I notice that there are a lot of paths through this code. I'm concerned that not all of them may be exercised by tests. If this piece of functionality retains a significant amount of complexity after you switch to `MultiConfig`, I'd recommend that you run the tests at least once with Coverage and make sure that all the paths through the code are covered.